tiatoolbox/visualization/tileserver.py

```diff
@@ -47,7 +51,39 @@ def __init__(self, title: str, layers: Dict[str, WSIReader]) -> None:
             static_folder=data._local_sample_path(Path("visualization") / "static"),
         )
         self.tia_title = title
-        self.tia_layers = layers
+
+        # generic layer names if none provided
+        if isinstance(layers, list):
+            layers = {f"layer-{i}": p for i, p in enumerate(layers)}
+        # set up the layer dict
+        layer_def = {}
+        meta = None
+        for i, key in enumerate(layers.keys()):
+            layer = layers[key]
+
+            if isinstance(layer, (str, Path)):
+                p = Path(layer)
+                if p.suffix in [".jpg", ".png"]:
+                    # assume its a low-res heatmap
+                    layer = Image.open(p)
+                    layer = np.array(layer)
+                else:
+                    layer = WSIReader.open(p)
+
+            if isinstance(layer, np.ndarray):
+                if len(layer.shape) == 2:
+                    # single channel, make into rgb with colormap
+                    c_map = cm.get_cmap("jet")
+                    im_rgb = (c_map(layer) * 255).astype(np.uint8)
+                    layer = VirtualWSIReader(im_rgb[:, :, :3], info=meta)
+                else:
+                    layer = VirtualWSIReader(p, info=meta)
```

Comment:
This would section could possibly be refactored into a function e.g. `colourise_image` which takes an ndarray and applies a cmap if is is a single channel array, else returns the image unmodified. Then this could be wrapped in VirtualWSIReader. E.g.

```python
if isinstance(layer, (str, Path)):
    p = Path(layer)
      if p.suffix in [".jpg", ".png"]:  # maybe add .webp and .gif here?
          # assume its a low-res heatmap
            layer = Image.open(p)
            layer = np.array(layer)
            layer = colourised_layer(layer, cmap=...)
            layer = VirtualWSIReader(layer, info=meta)
      else:
           layer = WSIReader.open(p)
```

I think this would make it a bit more readable / maintainable. This also does not account for low res TIFFs which we may need to address later.

Reply:
moved the colourising code into a function in utils/visualisation. Haven't added any additional formats for now, can maybe look at that later